lib/src/front_end/piece_factory.dart

```diff
@@ -604,10 +613,37 @@ mixin PieceFactory {
     });
   }
 
+  /// If [parameter] has a [defaultValue] then writes a piece for the parameter
+  /// followed by that default value.
+  ///
+  /// Otherwise, just writes [parameter].
+  void writeDefaultValue(
+      Piece parameter, (Token separator, Expression value)? defaultValue) {
+    if (defaultValue case (var separator, var value)) {
+      var operatorPiece = pieces.build(() {
+        if (separator.type == TokenType.EQ) pieces.space();
+        pieces.token(separator);
+        if (separator.type != TokenType.EQ) pieces.space();
+      });
+
+      var valuePiece = nodePiece(value, context: NodeContext.assignment);
+
+      pieces.add(AssignPiece(
+          left: parameter,
+          operatorPiece,
+          valuePiece,
+          canBlockSplitRight: value.canBlockSplit));
+    } else {
+      // No default value.
+      pieces.add(parameter);
+    }
```

Comment:
Consider refactoring to a guard clause and reducing indentation on the interesting path.

```suggestion
    if (defaultValue == null) {
      pieces.add(parameter);
      return;
    }
    var (separator, value) = defaultValue;
    var operatorPiece = pieces.build(() {
      if (separator.type == TokenType.EQ) pieces.space();
      pieces.token(separator);
      if (separator.type != TokenType.EQ) pieces.space();
    });

    var valuePiece = nodePiece(value, context: NodeContext.assignment);

    pieces.add(AssignPiece(
        left: parameter,
        operatorPiece,
        valuePiece,
        canBlockSplitRight: value.canBlockSplit));
```

Reply:
Great idea, thanks!